File: code/functions/pyprism/pyprism/data/_load_dsg.py
# -*- coding: utf-8 -*-
"""
@created: Feb 10 2021
@modified: Apr 25 2021
@author: Yoann Pradat

    CentraleSupelec
    MICS laboratory
    9 rue Juliot Curie, Gif-Sur-Yvette, 91190 France

    Institut Gustave Roussy
    Prism Center
    114 rue Edouard Vaillant, Villejuif, 94800 France

Functions
    load_dsg()
"""

from ._filepaths import _get_filepath_dsg_curated
from ._util_load import load_from_data
from pyprism.util import setwd_to_data

# PRISM ================================================================================================================

def load_dsg(study: str):
    """
    Load the design of the specified study.

    Parameters
    ----------
    study: str
        Only 'prism' is supported

    Returns
    -------
    df_des: DataFrame
    """
    if study.lower() == "prism" or study.lower() == "met500":
        df = load_from_data(_get_filepath_dsg_curated(study))
        if study=="prism":
            # Keep 1 FASTQ per sample
            df = df.dropna(subset=["Datafile_Path"])
            if "Comment_Sample(Y.Pradat)" in df:
                df = df[df["Comment_Sample(Y.Pradat)"].isnull()]
            assert df.shape[0]==df["Sample_Id"].nunique()
        return df
    else:
        raise ValueError("Unsupported value '%s' of study. Choose 'prism' or 'met500'.")

File: code/functions/rprism/R/load_dsg.R
# @created: 08 Apr 21
# @modified: 27 Apr 21
# @author: Yoann Pradat
# 
# Exported functions
#     load_dsg

#' Load the design table
#'
#' Load the design table for the specified study. Only "prism" is supported.
#'
#' @return a data.frame
#' @param study Name of the cohort used in the naming of the files. Choose "met500", "prism" or "tcga".
#'
#' @author Yoann Pradat
#'
#' @export
load_dsg <- function(study){
  load_from_data(get_filepath_dsg_curated(study), progress=F, show_col_types=F)
}
